### ISSUE TITLE
Getting `'clockOffset' of undefined` when following the Getting Started guide

### ISSUE THREAD

**Issue by cowchimp** [nearform/node-clinic-doctor#297]
## Expected Behavior
Running the example from the "Getting Started" section should work successfully.

## Current Behavior
Getting `'clockOffset' of undefined` error when following the "Getting Started" example

<details>
  <summary>Cannot read property 'clockOffset' of undefined</summary>

```
➜  clinic doctor --autocannon [ / ] -- node ./slow-io
Running 10s test @ http://localhost:3000/
10 connections

┌─────────┬────────┬─────────┬─────────┬─────────┬────────────┬───────────┬────────────┐
│ Stat    │ 2.5%   │ 50%     │ 97.5%   │ 99%     │ Avg        │ Stdev     │ Max        │
├─────────┼────────┼─────────┼─────────┼─────────┼────────────┼───────────┼────────────┤
│ Latency │ 711 ms │ 1304 ms │ 1861 ms │ 1887 ms │ 1315.23 ms │ 341.29 ms │ 1887.37 ms │
└─────────┴────────┴─────────┴─────────┴─────────┴────────────┴───────────┴────────────┘
┌───────────┬───────┬───────┬────────┬─────────┬─────────┬───────┬───────┐
│ Stat      │ 1%    │ 2.5%  │ 50%    │ 97.5%   │ Avg     │ Stdev │ Min   │
├───────────┼───────┼───────┼────────┼─────────┼─────────┼───────┼───────┤
│ Req/Sec   │ 2     │ 2     │ 8      │ 9       │ 7       │ 2.05  │ 2     │
├───────────┼───────┼───────┼────────┼─────────┼─────────┼───────┼───────┤
│ Bytes/Sec │ 300 B │ 300 B │ 1.2 kB │ 1.35 kB │ 1.05 kB │ 307 B │ 300 B │
└───────────┴───────┴───────┴────────┴─────────┴─────────┴───────┴───────┘

Req/Bytes counts sampled once per second.

70 requests in 10.06s, 10.5 kB read
[    ] Analysing data/Users/cowchimp/temp/node-clinic-doctor-examples-master/node_modules/@nearform/doctor/format/trace-event-decoder.js:49
          self.clockOffset = systemInfo.clockOffset
                                        ^

TypeError: Cannot read property 'clockOffset' of undefined
    at /Users/cowchimp/temp/node-clinic-doctor-examples-master/node_modules/@nearform/doctor/format/trace-event-decoder.js:49:41
    at Endpoint.finish (/Users/cowchimp/temp/node-clinic-doctor-examples-master/node_modules/endpoint/endpoint.js:36:5)
    at Object.onceWrapper (events.js:281:20)
    at Endpoint.emit (events.js:198:15)
    at finishMaybe (_stream_writable.js:646:14)
    at endWritable (_stream_writable.js:663:3)
    at Endpoint.Writable.end (_stream_writable.js:594:5)
    at onend (/Users/cowchimp/temp/node-clinic-doctor-examples-master/node_modules/duplexify/node_modules/readable-stream/lib/_stream_readable.js:627:10)
    at processTicksAndRejections (internal/process/task_queues.js:79:9)
```
</details>

## Steps to Reproduce (for bugs)

1. Test against `node-clinic-doctor-examples` as detailed here https://clinicjs.org/documentation
```
git clone git@github.com:nearform/node-clinic-doctor-examples.git
cd node-clinic-doctor-examples
npm install
clinic doctor --autocannon [ / ] -- node ./slow-io
```

## Environment
* Clinic.js version:
Clinic 5.0.0 and doctor 4.1.0
* Node.js version:
Node 11.14.0 and npm 6.9.0
* Browser name and version:
Chrome 78.0.3904.97
* Operating system and version:
OS X 10.13.6

**Comment by goto-bus-stop** [comment]
Transferred to the Doctor repo. I think it is a race condition. `format/trace-events-reader.js` tries to read the system info object from the `systemInfoReader` stream, but it might do this _after_ the object has already been emitted. If we do it inside the `TraceEventsDecoder` _constructor_ instead of the `_transform` method, we should be good

**Comment by goto-bus-stop** [comment]
Just got this on CI too: https://dev.azure.com/node-clinic/node-clinic/_build/results?buildId=2331&view=logs&j=9473ad5c-6f95-59ee-11bd-9c05c464119a&t=8edc8782-b559-5d0a-428e-c4998054f736

#302 shouuuuuuld fix it.

**Comment by goto-bus-stop** [comment]
This should be fixed in doctor :package: 4.1.1. Thanks!
